### PR TITLE
ResStock-HPXML: convert UpgradeCosts measure to ModelMeasure, try 2

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Supersedes https://github.com/NREL/buildstockbatch/pull/455.

## Checklist

Not all may apply

- [ ] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [ ] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/coverage.yml` as necessary.
- [ ] All other unit and integration tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run on Kestrel/Eagle to make sure it all works if you made changes that will affect Kestrel/Eagle
- [ ] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: buildstockbatch/workflow_generator/residential/latest/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+__version__ = "2024.07.19"
+version_info = {
+    "version": __version__,
+    "minimum_resstock_version": "3.2.0",
```
Comment (rajeee) [inline buildstockbatch/workflow_generator/residential/latest/__init__.py]: Should update this version to 3.3.0 after the res-reporting branch updates the ResStock version in buildstock.rb to 3.3.0.

File: buildstockbatch/workflow_generator/residential/latest/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+__version__ = "2024.07.19"
+version_info = {
+    "version": __version__,
+    "version_description": """
+Version information for the residential workflow generator.
+This version is compatible with ResStock v3.2.0 and later.
```
Comment (joseph-robertson) [inline buildstockbatch/workflow_generator/residential/latest/__init__.py]: Not true, right? Compatible with ResStock > v3.2.0.

File: buildstockbatch/test/test_inputs/test_openstudio_buildstock/resources/buildstock.rb [hunk 3]
```diff
@@ -1,8 +1,7 @@
 # frozen_string_literal: true
 class Version
-  ResStock_Version = '3.1.0' # Version of ResStock
+  ResStock_Version = '2000.1.0' # Version of ResStock
```
Comment (joseph-robertson) [inline buildstockbatch/test/test_inputs/test_openstudio_buildstock/resources/buildstock.rb]: Test against the new `WorkflowGenerator_Version` line?

File: docs/changelog/changelog_dev.rst [hunk 4]
```diff
@@ -78,3 +78,10 @@ Development Changelog
         :pullreq: 456
 
         Refactor WorkflowGenerator.
+
+    .. change::
+        :tags: general, feature
+        :pullreq: 458
+
+        For the Residential HPXML Workflow Generator, supports the UpgradeCosts measure
+        changing from ReportingMeasure to ModelMeasure.
```
Comment (joseph-robertson) [inline docs/changelog/changelog_dev.rst]: Probably worth a much more comprehensive changelog entry than just this.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -23,7 +23,7 @@ jobs:
         with:
           repository: NREL/resstock
           path: resstock
-          ref: develop
+          ref: res-reporting
```
Comment (joseph-robertson) [inline .github/workflows/ci.yml]: Before merge, need to switch this and down below back to develop (and make sure tests are still passing).

File: buildstockbatch/workflow_generator/residential/latest/__init__.py (hunk 2 above)
Comment (rajeee) [reply buildstockbatch/workflow_generator/residential/latest/__init__.py]: Correct. Thanks for catching this.

File: buildstockbatch/test/test_inputs/test_openstudio_buildstock/resources/buildstock.rb (hunk 3 above)
Comment (rajeee) [reply buildstockbatch/test/test_inputs/test_openstudio_buildstock/resources/buildstock.rb]: Hmm, I forget why I made this change. ResStock_Version doesn't really matter for BuildStockBatch. We only care about "WorkflowGenerator_Version". I will address this.

**Review by rajeee — APPROVED** [review]

**Pull request merged**